Safe Haskell-related build error

When building with a fresh checkout and apparently an old version of th-abstraction, I get the following error:

```
Building library for parameterized-utils-2.1.5.0.99..
[20 of 38] Compiling Data.Parameterized.ClassesC

src/Data/Parameterized/ClassesC.hs:32:1: error:
    Data.Parameterized.Some: Can't be safely imported!
    The module itself isn't safe.
   |
32 | import Data.Parameterized.Some (Some(..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I narrowed it down to a reproducible case with the following `cabal.project.freeze`:

```
active-repositories: hackage.haskell.org:merge
constraints: parameterized-utils +unsafe-operations,
             any.th-abstraction ==0.3.2.0

index-state: hackage.haskell.org 2022-07-13T02:14:33Z
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -48,8 +48,8 @@ jobs:
           echo GHC=$GHC >> $GITHUB_ENV
           case ${{ matrix.ghc }} in
             8.6.5) GHC_NIXPKGS=github:nixos/nixpkgs/20.09 ;;
-            9.0.2) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-unstable ;;
-            9.2.2) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-unstable ;;
+            9.0.2) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-22.05 ;;
+            9.2.2) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-22.05 ;;
             *)     GHC_NIXPKGS=github:nixos/nixpkgs/21.11 ;;
           esac
           echo NS="nix shell ${GHC_NIXPKGS}#${GHC}" >> $GITHUB_ENV

--- a/parameterized-utils.cabal
+++ b/parameterized-utils.cabal
@@ -51,7 +51,7 @@ library
   import: bldflags
   build-depends: base >= 4.10 && < 5
                , base-orphans   >=0.8.2 && <0.9
-               , th-abstraction >=0.3  && <0.5
+               , th-abstraction >=0.4.2 && <0.5
                , constraints    >=0.10 && <0.14
                , containers
                , deepseq

